controller/Review.php

```diff
@@ -136,6 +161,110 @@ public function index(): void
         $this->composeView('delegated-view', $data, 'pages/index.tpl', 'tao');
     }
 
+    private function pauseConcurrentSessions(
```

Comment:
This smells like a complete new Service, not a controller method. Wdyt?

Reply:
I though about that, but eventually found the changes to be too cumbersome at that time and went back to keep the logic in the controller. The problem lies with the later call to store the pause reason as part of the user session, which would bound the service to the user session (and thus making it stateful).

```php
 $this->setSessionAttribute(
            "pauseReason-{$execution->getIdentifier()}",
            PauseService::PAUSE_REASON_CONCURRENT_TEST
        );
```

I'll try again to move it into a service (probably returning the IDs for the paused executions so the controller can still store the pause reason for each test in the user session).

Reply:
This should make it easier to move the code to pause all concurrent sessions into a service: 216481426b3c2b5fd5654344cb09bdd058e0c4e4